includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -1045,7 +1045,11 @@ private function validate_subscription_data( $subscription_data ) {
 	 * @return bool True if store has active WCPay subscriptions, otherwise false.
 	 */
 	public static function store_has_active_wcpay_subscriptions() {
-		$results = wcs_get_subscriptions(
+		if ( ! function_exists( 'wcs_get_subscriptions' ) ) {
```

Comment:
So after testing the various plugin active/deactivated, feature enabled/disabled and migration enabled/disabled I realised that this change was made because if you have:

- WC Subscriptions deactivated
- the WCPay Subscriptions feature deactivated
- but the migration feature enabled

then this function is called during the `WC_Payments_Subscriptions::init()` flow (via `WC_Payments_Subscriptions_Plugin_Notice_Manager` class) and in that environment no Subscriptions functionality is being loaded hence an error.

I think that might indicates a broader issue.

I wonder if we need to change the 

```
if ( WC_Payments_Features::is_wcpay_subscriptions_enabled() || WC_Payments_Features::is_subscription_migration_enabled() ) {
```

logic to instead check

1. Is it wcpay subscriptions feature enabled
2. or migration enabled and plugin active. 

🤔

Reply:
Thanks @james-allan for the initial review of this.

I've made a few changes to this PR following this feedback:
 - [8718ad9](https://github.com/Automattic/woocommerce-payments/pull/6942/commits/8718ad901a48fd840234d3e4cd78889d13d3a847) Moved loading the migrator to be inside the main WCPay Subscriptions class
 - [844c4d0](https://github.com/Automattic/woocommerce-payments/pull/6942/commits/844c4d08d07e37f6dac4a5a04888503abc1cbfaa) Initialise the main WCPay Subscriptions class when the WCPay Subscriptions feature is enabled OR the migration feature is enabled and WC Subscriptions is active.
 - [30177b0](https://github.com/Automattic/woocommerce-payments/pull/6942/commits/30177b0b7b44f68ed9e471f9e8500e2736de8ed3) Because of the above change, I went through each WCPay subscription class and added any missing checks to make sure we're not running any unexpected code now that these classes will be loaded when migration + WC Subscriptions is active.
 - Some smaller changes to fix unit tests and make sure the new `_migrated_` meta is properly set and not being copied to subscription related orders by the WCS copier.

I've ran through some basic tests with these latest changes to ensure no fatal errors but we'll need to do more thorough testing and/or code review to make sure I'm not introducing a bug here.